src/lib/es2019.array.d.ts

```diff
@@ -1,3 +1,10 @@
+type Flat<A, D extends number> = {
+    '1': A,
+    '0': A extends ReadonlyArray<infer T>
+        ? Flat<T, [-1, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20][D]>
+        : A
+}[D extends -1 ? '1' : '0']

```

Comment:
'1' and '0' should really have meaningful names since they're not used as numeric indices. I would call them 'done' and 'recur' or something like that.

Reply:
Renamed to `done` and `recur`